python/src/scipp/_shape.py

```diff
@@ -47,17 +49,46 @@ def concatenate(x, y, dim):
     :raises: If the dtype or unit does not match, or if the
              dimensions and shapes are incompatible.
     :return: The absolute values of the input.
+
+    Examples:
+
+      >>> a = sc.arange('x', 3)
+      >>> b = 100 * sc.arange('x', 3)
+      >>> c = sc.concatenate(a, b, 'x')
+      >>> c
+      <scipp.Variable> (x: 6)      int64  [dimensionless]  [0, 1, ..., 100, 200]
```

Comment:
Is it a good idea to include inputs here? These may change, and no one will remember (or want to) update all the docstrings by hand. Is there a way to auto-generate?

Reply:
Do you mean 'outputs'? There is no way to understand the examples without inputs. Concerning outputs, don't know if they can be generated.

I based this on the numpy and xarray docs which both seem to do it this way. Numpy is actually quite bad, they have loads of repetition.

Do you have an idea how this can be done? We cannot cover all functions in notebooks. Especially when we want to use autodoc.

Reply:
Yes, I meant outputs.

No, I don't have an idea. In fact I was looking into something similar previously and learned that it is *not* possible to generate docstrings (e.g., by having a function that returns a string).

I suppose the alternative would be to provide a separate notebook or `rst` file that contains the examples? This would actually have a significant advantage: We would have builds fails indicating examples that are outdated. Mantid, for example, relies on this heavily: Every algorithm has a `rst` with code examples that are excuted as part of the docs builds.

Reply:
Can those examples be merged into the generated documentation? Or would we have to drop docstrings and put everything into separate files?

Reply:
I am not aware of a way for merging. We'd just have to put a link into the docstring?